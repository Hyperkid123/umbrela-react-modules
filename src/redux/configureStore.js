import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
