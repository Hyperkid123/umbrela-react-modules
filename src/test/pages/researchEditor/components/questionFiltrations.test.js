import React from 'react'
import QuestionFiltrations, {QuestionFiltrations as Snapshot} from '../../../../pages/researchEditorModule/components/questionFiltrations';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Question filtrations', () => {
  const initialState  = {
    options: {
      allOptions: {
        1: []
      }
    },
    editor: {
      sheets: []
    },
    activeSheet: {}
  };

  const mockStore = configureStore();
  it('should create component correctly', () => {
    let store = mockStore(initialState);
    let wrapper = shallow(<QuestionFiltrations questionId={1} store={store}/>);
    expect(wrapper).toBeDefined();
  });

  it('should render correctly empty', () => {
    const tree = shallow(
      <Snapshot loadOptions={jest.fn()} />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should render sheets list', () => {
    const tree = shallow(
      <Snapshot
        loadOptions={jest.fn()}
        activeSheet={{sheetOrder: 0, sheetId: 10}}
        sheets={[
          {sheetOrder: 1, sheetId: 10, title: 'foo'}
        ]}
        questionId={10}
        questionOrder={10}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should set tab correctly', () => {
    const options = [{
      optionId: 0
    }, {
        optionId: 10
    }];
    const wrapper = shallow(
      <Snapshot
        options={options}
        loadOptions={jest.fn()}
      />
    );
    wrapper.instance().setCurrentTab(1);
    expect(wrapper.state().tab).toEqual(1);
    expect(wrapper.state().selectedOptionId).toEqual(options[1].optionId);
  });

  it('should load options on question change', () => {
    const loadOptions = jest.fn();
    const wrapper = shallow(
      <Snapshot
        options={[]}
        loadOptions={loadOptions}
        questionId={0}
      />
    );
    expect(loadOptions.mock.calls.length).toEqual(1);
    wrapper.setProps({questionId: 1});
    expect(loadOptions.mock.calls.length).toEqual(2);
    wrapper.setProps({questionId: 1});
    expect(loadOptions.mock.calls.length).toEqual(2);
  });

  it('should set first atb if new options are loaded', () => {
    const wrapper = shallow(
      <Snapshot
        options={false}
        loadOptions={jest.fn()}
        questionId={0}
      />
    );
    wrapper.setProps({options: [{optionId: 99}]});
    expect(wrapper.state().selectedOptionId).toEqual(99);
  });

  it('shoud not render question list', () => {
    const tree = shallow(
      <Snapshot
        loadOptions={jest.fn()}
        activeSheet={{sheetOrder: 50, sheetId: 1}}
        sheets={[
          {sheetOrder: 0, sheetId: 10, title: 'foo'}
        ]}
        questionId={10}
        questionOrder={10}
      />
    ).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });
});
