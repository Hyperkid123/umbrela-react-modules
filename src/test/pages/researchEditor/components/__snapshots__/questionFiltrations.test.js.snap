// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question filtrations shoud not render question list 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles(AppBar)
    color="default"
    position="static"
    style={
      Object {
        "maxWidth": "calc(100vw - 400px - 20px)",
      }
    }
  >
    <WithStyles(Tabs)
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      scrollButtons="auto"
      scrollable={true}
      value={0}
    />
  </WithStyles(AppBar)>
  <styled.div
    horizontalCenter={true}
    wrap="true"
  />
</div>
`;

exports[`Question filtrations should render correctly empty 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles(AppBar)
    color="default"
    position="static"
    style={
      Object {
        "maxWidth": "calc(100vw - 400px - 20px)",
      }
    }
  >
    <WithStyles(Tabs)
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      scrollButtons="auto"
      scrollable={true}
      value={0}
    />
  </WithStyles(AppBar)>
  <styled.div
    horizontalCenter={true}
    wrap="true"
  />
</div>
`;

exports[`Question filtrations should render sheets list 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <WithStyles(AppBar)
    color="default"
    position="static"
    style={
      Object {
        "maxWidth": "calc(100vw - 400px - 20px)",
      }
    }
  >
    <WithStyles(Tabs)
      fullWidth={true}
      indicatorColor="primary"
      onChange={[Function]}
      scrollButtons="auto"
      scrollable={true}
      value={0}
    />
  </WithStyles(AppBar)>
  <styled.div
    horizontalCenter={true}
    wrap="true"
  >
    <Connect(FilterableQuestionsList)
      filterOrder={true}
      key="10"
      questionId={10}
      questionOrder={10}
      selectedOptionId={null}
      sheetId={10}
      title="foo"
    />
  </styled.div>
</div>
`;
